Azure/azure-cli-extensions
# {AKS} Sort out tests and update default test matrix

---

This checklist is used to make sure that common guidelines for a pull request are followed.

Sort out tests and update default test matrix.

- Remove all test cases that are duplicated with test cases in azure-cli/acs and their corresponding recording files (if exist)
  - test_aks_create_aadv1_and_update_with_managed_aad
  - test_aks_enable_utlra_ssd
  - test_aks_create_with_managed_disk
  - test_aks_disable_addons_confcom_addon
  - test_aks_create_with_windows
  - test_aks_byo_subnet_with_ingress_appgw_addon
  - test_aks_nodepool_update_label_msi
  - test_aks_create_addon_with_azurekeyvaultsecretsprovider_with_secret_rotation
  - test_aks_create_nonaad_and_update_with_managed_aad
  - test_aks_update_with_windows_password
  - test_aks_create_with_azurekeyvaultsecretsprovider_addon
  - test_aks_custom_kubelet_identity
  - test_aks_stop_and_start
  - test_aks_upgrade_node_image_only_nodepool
  - test_aks_create_private_cluster_public_fqdn
  - test_aks_create_with_confcom_addon
  - test_aks_update_to_msi_cluster
  - test_aks_disable_addon_openservicemesh
  - test_aks_byo_appgw_with_ingress_appgw_addon
  - test_aks_create_with_ahub
  - test_aks_create_with_confcom_addon_helper_enabled
  - test_aks_create_with_ingress_appgw_addon
  - test_aks_create_with_openservicemesh_addon
  - test_aks_create_with_ephemeral_disk
  - test_aks_create_with_auto_upgrade_channel
  - test_aks_create_fqdn_subdomain
  - test_aks_create_with_ossku
  - test_aks_enable_addons_confcom_addon
  - test_aks_update_label_msi
  - test_aks_disable_local_accounts
  - test_aks_enable_addon_with_azurekeyvaultsecretsprovider
  - test_aks_enable_addon_with_openservicemesh
  - test_aks_create_and_update_with_managed_aad_enable_azure_rbac
  - test_aks_create_and_update_with_managed_aad
  - test_aks_nodepool_add_with_ossku
  - test_aks_upgrade_node_image_only_cluster
- Update default test matrix
  - Add gmsa related test cases back since now it supports `--aks-custom-headers`
  - Add openservicemesh and keyvault secret provider related test cases back since they no longer rely on feature registration.
  - Add ossku related test cases back since the compatibility issue related to CBL-Mariner has been resolved.

### General Guidelines


For new extensions:



### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.
